packages/assets-controllers/src/NftController.test.ts

```diff
@@ -3435,4 +3436,229 @@ describe('NftController', () => {
+        tokenId: '3',
+        standard: 'ERC721',
+        favorite: false,
+        isCurrentlyOwned: true,
+        tokenURI: 'https://api.pudgypenguins.io/lil/4',
+      });
+    });
+
+    it('should not update metadata when calls to fetch metadata fail', async () => {
+      const { nftController } = setupController();
+      const { selectedAddress } = nftController.config;
+      const spy = jest.spyOn(nftController, 'updateNft');
+      const testNetworkClientId = 'sepolia';
+      await nftController.addNft('0xtest', '3', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      sinon
+        // TODO: Replace `any` with type
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .stub(nftController, 'getNftInformation' as any)
+        .rejects(new Error('Error'));
+      const testInputNfts: Nft[] = [
+        {
+          address: '0xtest',
+          description: null,
+          favorite: false,
+          image: null,
+          isCurrentlyOwned: true,
+          name: null,
+          standard: 'ERC721',
+          tokenId: '3',
+        },
+      ];
+
+      await nftController.updateNftMetadata(testInputNfts, testNetworkClientId);
+
+      expect(spy).toHaveBeenCalledTimes(0);
+      expect(
+        nftController.state.allNfts[selectedAddress][SEPOLIA.chainId][0],
+      ).toStrictEqual({
+        address: '0xtest',
+        description: '',
+        favorite: false,
+        image: '',
+        isCurrentlyOwned: true,
+        name: '',
+        standard: 'ERC721',
+        tokenId: '3',
+      });
+    });
+
+    it('should  update metadata when some calls to fetch metadata succeed', async () => {
+      const { nftController } = setupController();
+      const { selectedAddress } = nftController.config;
+      const spy = jest.spyOn(nftController, 'updateNft');
+      const testNetworkClientId = 'sepolia';
+      // Add nfts
+      await nftController.addNft('0xtest1', '1', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      await nftController.addNft('0xtest2', '2', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      await nftController.addNft('0xtest3', '3', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      sinon
+        // TODO: Replace `any` with type
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .stub(nftController, 'getNftInformation' as any)
```

Comment:
`any` is unnecessary here since the compiler tells us exactly what the `as` assertion target type should be

> Argument of type '"getNftInformation"' is not assignable to parameter of type 'keyof NftController'.ts(2345)

```suggestion
        .stub(nftController, 'getNftInformation' as keyof typeof nftController)
```

In general, shared libraries is trying to avoid introducing new `any` types except under a [few very limited circumstances](https://github.com/MetaMask/contributor-docs/pull/74/files#diff-ae03a9f2eb638a25c1d8d7ba853c78220f1ed44a19ad97a8d780cbc0c00a4260R318-R401). An `as` assertion even to an approximate type is [always preferred](https://github.com/MetaMask/contributor-docs/pull/74/files#diff-ae03a9f2eb638a25c1d8d7ba853c78220f1ed44a19ad97a8d780cbc0c00a4260R195-R199).